packages/cadl-ranch-specs/http/server/version/union/main.tsp

```diff
@@ -0,0 +1,66 @@
+import "@typespec/http";
+import "@typespec/versioning";
+import "@azure-tools/cadl-ranch-expect";
+import "@azure-tools/typespec-azure-core";
+
+using Azure.Core;
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+
+@doc("Illustrates server with customer-defined and template-defined api versions.")
+@scenarioService("/server/version/union")
+@useDependency(Azure.Core.Versions.v1_0_Preview_1)
+namespace Server.Version.Union;
```

Comment:
```suggestion
namespace Server.Version.Union;
```

What is the meaning of `union` here? what exactly do you expect a client to be doing with this spec in the generated code(not the url sent)

Reply:
`Union` is copied from `authentication/union`, I'm OK if you have a better name. This is a real case, that customer would use template from Azure.Core and define their operation with their own api-version in the same spec file.